### PR TITLE
add tokenreview to support login check for headers

### PR DESCRIPTION
**Pull request by mlbiam**
additional capability for https://github.com/kiali/kiali/issues/3207

### CODE REVIEW

File: kiali-server/templates/role.yaml [hunk 1]
```diff
@@ -99,4 +99,10 @@ rules:
   - list
   - patch
   - watch
+- apiGroups:
+  - authentication.k8s.io
```
Comment (jmazzitelli) [inline kiali-server/templates/role.yaml:103]: Maintain consistency with the rest of this file - specify apiGroups using `[ ]` notation.
```suggestion
- apiGroups: ["authentication.k8s.io"]
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
minor change request

File: kiali-server/templates/role.yaml (hunk 1 above)
Comment (mlbiam) [reply kiali-server/templates/role.yaml]: done

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Because the server now requires a new permission, the operator has to be given this permission also (because the operator will be creating Roles for Kiali and in order to give a permission one has to already have that permission first).

So, this permission needs to also be added to the operator helm chart as well.

So you will need to add this to the bottom here: https://github.com/kiali/helm-charts/blob/master/kiali-operator/templates/clusterrole.yaml#L297

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Also missing the permission in the role_viewer template.

Read the DEVELOPING.adoc section on "[Are you altering a kiali server role permission](https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-role-permission)" for details on all the things you need to do when adding a permission.

**Review by jmazzitelli — DISMISSED** [review]
As per [checklist](https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-role-permission), you are still missing this:
* Update the [Server Helm Chart role-viewer.yaml](https://github.com/kiali/helm-charts/tree/master/kiali-server/templates/role-viewer.yaml)

**Review by jmazzitelli — APPROVED** [review]
LGTM

**Pull request merged**